src/main/app/forms/models/partyDetails.ts

```diff
@@ -23,48 +32,51 @@ export class PartyDetails implements Serializable<PartyDetails>, CompletableTask
   correspondenceAddress?: CorrespondenceAddress
 
   constructor (
+    name?: string,
     address: Address = new Address(),
     hasCorrespondenceAddress: boolean = false,
     correspondenceAddress: Address = new CorrespondenceAddress()
   ) {
     this.address = address
     this.hasCorrespondenceAddress = hasCorrespondenceAddress
     this.correspondenceAddress = correspondenceAddress
+    this.name = name
   }
 
   static fromObject (input?: any): PartyDetails {
     if (input == null) {
       return input
     }
-
-    let deserialized: PartyDetails = new PartyDetails(
+    const deserialized: PartyDetails = new PartyDetails(
+      input.name,
       new Address().deserialize(input.address),
       input.hasCorrespondenceAddress === 'true',
       new CorrespondenceAddress().deserialize(input.correspondenceAddress)
     )
-
     if (deserialized.hasCorrespondenceAddress === false) {
       deserialized.correspondenceAddress = undefined
     }
-
+    deserialized.type = input.type
     return deserialized
   }
 
-  deserialize (input: any): PartyDetails {
+  deserialize (input?: any): PartyDetails {
     if (input) {
       this.address = new Address().deserialize(input.address)
-      this.hasCorrespondenceAddress = input.hasCorrespondenceAddress
+      this.type = input.type
+      this.name = input.name
+      this.address = new Address().deserialize(input.address)
+      this.hasCorrespondenceAddress = input.hasCorrespondenceAddress === 'true' || input.hasCorrespondenceAddress === true
```

Comment:
Having `input.hasCorrespondenceAddress === 'true'` smells bed, instead of this deserialisation should have been implemented.

ACTION NEEDED

Reply:
Removed. A `deserialize` should not do any conversion.